src/Draco.Compiler/Internal/Declarations/DeclarationTable.cs

```diff
@@ -15,46 +19,100 @@ internal sealed class DeclarationTable
     /// Constructs a new declaration table from the given syntax trees.
     /// </summary>
     /// <param name="syntaxTrees">The syntax trees to construct the declarations from.</param>
+    /// <param name="compilation">The compilation using this declaration table.</param>
     /// <returns>The declaration table containing <paramref name="syntaxTrees"/>.</returns>
-    public static DeclarationTable From(ImmutableArray<SyntaxTree> syntaxTrees) => new(syntaxTrees);
+    public static DeclarationTable From(ImmutableArray<SyntaxTree> syntaxTrees, Compilation compilation) => new(syntaxTrees, compilation);
 
     /// <summary>
     /// An empty declaration table.
     /// </summary>
-    public static DeclarationTable Empty { get; } = new(ImmutableArray<SyntaxTree>.Empty);
+    public static DeclarationTable Empty { get; } = new(ImmutableArray<SyntaxTree>.Empty, Compilation.Create(ImmutableArray<SyntaxTree>.Empty));
 
     /// <summary>
     /// The merged root module.
     /// </summary>
     public MergedModuleDeclaration MergedRoot => this.mergedRoot ??= this.BuildMergedRoot();
     private MergedModuleDeclaration? mergedRoot;
 
+    /// <summary>
+    /// The root path of this declaration table.
+    /// </summary>
+    public string RootPath => this.compilation.RootModulePath;
+
+    private readonly Compilation compilation;
     private readonly ImmutableArray<SyntaxTree> syntaxTrees;
 
-    private DeclarationTable(ImmutableArray<SyntaxTree> syntaxTrees)
+    private DeclarationTable(ImmutableArray<SyntaxTree> syntaxTrees, Compilation compilation)
     {
         this.syntaxTrees = syntaxTrees;
+        this.compilation = compilation;
     }
 
-    // NOTE: We don't have modules specified yet, so all added syntaxes are assumed to be in a global module with empty name
-    private MergedModuleDeclaration BuildMergedRoot() =>
-        new(this.syntaxTrees.Select(s => new SingleModuleDeclaration(string.Empty, (CompilationUnitSyntax)s.Root)).ToImmutableArray());
+    private MergedModuleDeclaration BuildMergedRoot()
+    {
+        // If we don't have root path, we put all file into top level module
+        if (string.IsNullOrEmpty(this.RootPath))
+        {
+            var singleModules = this.syntaxTrees
+                .Select(s => new SingleModuleDeclaration(string.Empty, string.Empty, (CompilationUnitSyntax)s.Root))
```

Comment:
I'd suggest using named args, otherwise it's hard to see what exactly `string.Empty` passed as